File: frontend/types/skill.ts
export interface Skill {
  workspace_id: string
  id: string
  slug: string
  name: string
  description: string
  instructions: string
  created_by: string
  created_at: string
  updated_at: string
}

export interface CreateSkillInput {
  name: string
  description: string
  instructions: string
}

export interface UpdateSkillInput {
  name?: string
  description?: string
  instructions?: string
}

File: backend/src/models/skill.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Skill {
    pub workspace_id: Uuid,
    pub id: Uuid,
    pub slug: String,
    pub name: String,
    pub description: String,
    pub instructions: String,
    pub created_by: Uuid,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Deserialize)]
pub struct CreateSkillRequest {
    pub name: String,
    pub description: String,
    pub instructions: String,
}

#[derive(Debug, Deserialize)]
pub struct UpdateSkillRequest {
    pub name: Option<String>,
    pub description: Option<String>,
    pub instructions: Option<String>,
}
